qiime/core/type/signature.py

```diff
@@ -73,6 +76,17 @@ def _assert_valid_parameters(self, parameters):
                     "Parameter %r must be a complete primitive type "
                     "expression, not %r" % (param_name, primitive_type))
 
+    def _assert_valid_defaults(self, defaults, parameters):
+        # only parameters can have defaults and the type must match
+        for param_name, default in defaults.items():
+            if param_name not in parameters:
+                raise ValueError("Input %r must not have a default value" %
+                                 param_name)
+
+            if not isinstance(default, parameters[param_name][1]):
```

Comment:
Good comment, thanks. I wasn't aware that you could use a semantic type in that way.